### PR TITLE
Fix ctail build by adding dependency on apr and apr-util

### PR DESCRIPTION
**Pull request by Moisan**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)? -- It fails as not being notable enough, however the formula was already into Homebrew.

---
- Add dependency on `apr` and `apr-util`
- Add their path to the configure flags
- Add a test

### CODE REVIEW

**Comment by Moisan** [comment]
This should also fix the Sierra bottle #5488 issue for ctail.

File: Formula/ctail.rb [hunk 1]
```diff
@@ -12,15 +13,21 @@ class Ctail < Formula
...
 
   def install
-    system "./configure", "--prefix=#{prefix}", "--disable-debug"
+    system "./configure", "--prefix=#{prefix}", "--disable-debug", "--with-apr=#{Formula["apr"].bin}", "--with-apr-util=#{Formula["apr-util"].bin}"
     system "make", "LIBTOOL=glibtool --tag=CC"
     system "make", "install"
   end
+
+  test do
+    system "ctail", "-h"
```
Comment (MikeMcQuaid) [inline Formula/ctail.rb]: Test is failing: https://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/8708/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_ctail/

Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

Finally, regardless it should be `system "#{bin}/ctail", "-h"`

File: Formula/ctail.rb [hunk 2]
```diff
@@ -12,15 +13,21 @@ class Ctail < Formula
     sha256 "dfab40d65950327c679bde97a335779526c58e99f5679f32f95e517a7249e332" => :mountain_lion
   end
 
-  conflicts_with "byobu", :because => "both install `ctail` binaries"
-
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "libtool" => :build
+  depends_on "apr" => :build
+  depends_on "apr-util" => :build
```
Comment (MikeMcQuaid) [inline Formula/ctail.rb]: Looks from the failing test that one or more of these are needed at runtime (i.e. no `:build`)

File: Formula/ctail.rb [hunk 3]
```diff
@@ -12,15 +13,21 @@ class Ctail < Formula
...
+
+  conflicts_with "byobu", :because => "both install `ctail` binaries"
 
   def install
-    system "./configure", "--prefix=#{prefix}", "--disable-debug"
+    system "./configure", "--prefix=#{prefix}", "--disable-debug", "--with-apr=#{Formula["apr"].bin}", "--with-apr-util=#{Formula["apr-util"].bin}"
```
Comment (MikeMcQuaid) [inline Formula/ctail.rb]: Use `.opt_bin` for these and please wrap onto multiple lines so it's not longer than 80 characters.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/ctail.rb (hunk 1 above)
Comment (Moisan) [inline Formula/ctail.rb]: I don't see how to create a  more substantial test: ctail needs an ssh connection to apply tail on a remote machine. I don't know how to mock such connection.

File: Formula/ctail.rb [hunk 4]
```diff
@@ -12,15 +13,25 @@ class Ctail < Formula
     sha256 "dfab40d65950327c679bde97a335779526c58e99f5679f32f95e517a7249e332" => :mountain_lion
   end
 
-  conflicts_with "byobu", :because => "both install `ctail` binaries"
-
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "libtool" => :build
+  depends_on "apr"
+  depends_on "apr-util"
```
Comment (MikeMcQuaid) [inline Formula/ctail.rb]: `depends_on :apr` should be sufficient here for all but Sierra and will work on Sierra when https://github.com/Homebrew/brew/pull/1227 is merged shortly. Sorry about the change, it's news to me too.

Comment (MikeMcQuaid) [inline Formula/ctail.rb]: CC @DomT4 @ilovezfs to check my work here.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by Moisan** [comment]
It worked on my machine but TestBot doesn't seem to find the `apr` files. Did I do something wrong?

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
https://github.com/Homebrew/brew/pull/1227 was merged and I didn't think the `revision` was needed so I've edited the formula and will pull when 💚.

**Comment by MikeMcQuaid** [comment]
We actually decided to change the https://github.com/Homebrew/brew/pull/1227 approach so I've edited again 😀

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**